content/en/blog/2023/basic-connector/index.md

```diff
@@ -0,0 +1,574 @@
+
+Setup the configuration you will use for the OpenTelemetry Collector in the
+config.yaml file. This file defines how your data will be routed, processed and
+exported. The configurations defined in the file, detail how you want your data
+pipeline to behave. You can define the components and how the data moves through
+your defined pipeline from start to end. There are further details about how to
+configure a Collector at
+[Collector Configurations](https://opentelemetry.io/docs/collector/configuration/).
+
+Use the following code for the example connector we will build. The code is an
+example of a basic valid OpenTelemetry Collector configuration file.
+
+    receivers:
+        otlp:
+            protocols:
+                grpc:
+                    endpoint: localhost:4317
+                http:
+                    endpoint: localhost:4318
+
+    exporters:
+        logging:
+
+    connectors:
+        example:
+
+    service:
+        pipelines:
+            traces:
+                receivers: [otlp]
+                processors: [batch]
+                exporters: [example]
+            metrics:
+                receivers: [example]
+                exporters: [logging]
+
+In the connectors portion of the above code, you need to declare the names of
+the usable connectors for your pipeline. Here, “example” is the name of the
+connector we will create in this tutorial.
+
+&nbsp;<br>
+
+## Implementation
+
+1.  Create a folder for your example connector. In this article we will create a
+    folder called exampleconnector.
+2.  Navigate to the folder and run
+
+        go mod init github.com/exampleconnector
+
+3.  Run `go mod tidy`
+
+    This will create go.mod and go.sum files.
+
+4.  Create the following files in the folder
+    - config.go - A file to define the connector's settings
+    - factory.go - A file to create instances of the connector
+
+&nbsp;<br>
+
+### Create your connector settings in config.go
+
+In order to be instantiated and participate in pipelines, the Collector needs to
+identify your connector and properly load its settings from within its
+configuration file.
+
+In order to be able to give your connector access to its settings, create a
+Config struct. The struct must have an exported field for each of the
+connector’s settings. The parameter fields added will be accessible from the
+config.yaml file. Their name in the configuration file is set through a struct
+tag. Create struct and add parameters. You can optionally add a validator
+function to check if the given default values are valid for an instance of your
+connector.
+
+    package exampleconnector
+
+
+    import "fmt"
+
+
+    type Config struct {
+        AttributeName string `mapstructure:"attributeName"`
+    }
+
+
+    func (c *Config) Validate() error {
+        if c.AttributeName != "request.n" {
+    	    return fmt.Errorf("the attribute for the data passed through the connector needs to be 'request.n' for this tutorial ")
+        }
```

Comment:
I would do something like this instead:
```suggestion
        if c.AttributeName == "" {
    	    return fmt.Errorf("attribute_name must not be empty")
        }
```

I believe the tutorial will still work if people change the attribute name in the appropriate places, and checking for it being empty is closer to something that is useful IRL